### PR TITLE
Move rules W2509, E2004, E2505, E2510 to E3030, E3031, E3008

### LINKED ISSUE #1084: extend CIDR regex validation
Rules [`W2509`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/rules/parameters/Cidr.py) and [`E2004`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/rules/parameters/CidrAllowedValues.py) check CIDR parameters have `AllowedPattern` and `AllowedValues` are CIDRs respectively

```yaml
Resources:
  ReceiptFilter:
    Type: AWS::SES::ReceiptFilter
    Properties:
      Filter:
        IpFilter:
          Policy: Block
          Cidr: 0.0.0.0/999
```

`Invalid CIDR block: 0.0.0.0/999 (Service: AmazonSimpleEmailService; Status Code: 400; Error Code: InvalidParameterValue; Request ID: )`

* Inline values
* Mappings
* Parameter Defaults
* evaluate AllowedPatterns

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*
Fix #1084 

*Description of changes:*
- Move rule E2510, W2509, E2505 and E2004 to E3030, E3031, E3008
- Add rule W2031 to validate AllowedPatternRegex on Parameter Default and AllowedValues attributes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types.json [hunk 1]
```diff
@@ -4,6 +4,33 @@
     "path": "/ValueTypes",
     "value": {
       "AvailabilityZone": {
+        "AllowedValues": [
```
Comment (PatMyron) [inline src/cfnlint/data/ExtendedSpecs/all/03_value_types.json:7]: We should remove this list here as part of this move:
https://github.com/aws-cloudformation/cfn-python-lint/blob/4eccd3faa7775c16521e2b5b827b7e8c1f9bffd4/src/cfnlint/helpers.py#L82

**Comment by PatMyron** [comment]
seems like [`W2509`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/docs/rules.md#W2509) violations were strictly lost for [`quickstart/`](https://github.com/aws-cloudformation/cfn-python-lint/tree/master/test/fixtures/templates/quickstart) templates rather than just switching rule IDs

**Comment by kddejong** [comment]
We specifically lost errors like this.  Which basically asks for a Parameter to have either AllowedValues or an AllowedPattern when doing a Cidr based parameter.  We could create a new rule that validates if an AllowedPattern is available and we are REFing a parameter that AllowedValues or AllowedPattern are provided.  Regex should match what we define.
```
"Message": "AllowedPattern and/or AllowedValues for Parameter should be specified at Parameters/pManagementCIDR. Example for AllowedPattern: '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\\/([0-9]|[1-2][0-9]|3[0-2]))$'"
```

I decided I'm not sure we want to do this.  Do you have thoughts on this?

**Comment by PatMyron** [comment]
[`E2004`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/docs/rules.md#E2004) seems similar as well

**Comment by kddejong** [comment]
> [`E2004`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/docs/rules.md#E2004) seems similar as well

Weird.  Yea I should have that covered in this PR too.

**Pull request merged**